freqtrade/persistence/trade_model.py

```diff
@@ -1205,6 +1205,35 @@ def calc_profit_ratio(
 
         return float(f"{profit_ratio:.8f}")
 
+    def calc_close_rate_for_roi(self, target_roi: float) -> float:
+        """
+        Calculate the required close price to reach a target ROI.
+        Matches the logic used in `calc_profit_ratio()`.
+
+        :param target_roi: The desired return on investment (as a decimal, e.g., 0.05 for 5%)
+        :return: Close price (rate) required to achieve the target ROI
+        """
+        leverage = float(self.leverage or 1.0)
+        deleveraged_roi = float(target_roi) / leverage
+
+        open_value = self._calc_open_trade_value(self.amount, self.open_rate)
+
+        # The ROI formula uses close_value(rate), which depends on trading mode:
+        # - SPOT: linear in rate, adjusted by close fee
+        # - MARGIN: same, but long subtracts interest, short increases amount
+        # - FUTURES: adds/subtracts funding to/from close value
+        # All cases are affine in rate:
+        #     close_value(rate) = a * rate + b
+        # We extract a and b by probing close_value at rate = 0 and 1.
+        value_at_0 = self.calc_close_trade_value(0.0)
+        value_at_1 = self.calc_close_trade_value(1.0)
```

Comment:
I'm not entirely sure i understand this point / section.

If you translate this into words, it's "close value at rate 0" - which would always be 100% loss i'd think? (except for a short - but let's ignore this for now).
Same with `calc_close_trade_value` - why would the value of the trade if price is 1 be relevant? 
A coin might be priced at 0.0001 - or at 100_000 - so what's "price at rate 1" exactly saying?